docs/examples/use_cases/tensorflow/efficientdet/pipeline/dali/efficientdet_pipeline.py

```diff
@@ -0,0 +1,262 @@
+                classes,
+                widths,
+                heights,
+                self._image_size,
+                [0.1, 2.0] if self._is_training else None,
+            )
+
+            if self._device == "gpu":
+                bboxes = bboxes.gpu()
+                classes = classes.gpu()
+
+            enc_bboxes, enc_classes = dali.fn.box_encoder(
+                bboxes, classes, anchors=self._boxes, offset=True, device=self._device
+            )
+            num_positives = dali.fn.reductions.sum(
+                dali.fn.cast(enc_classes != 0, dtype=dali.types.FLOAT),
+                device=self._device,
+            )
+            enc_classes -= 1
+
+            # convert to tlbr
+            enc_bboxes = dali.fn.coord_transform(
+                enc_bboxes,
+                M=[0, 1, 0, 0,
+                   1, 0, 0, 0,
+                   0, 0, 0, 1,
+                   0, 0, 1, 0],
+                device=self._device,
+            )
+
+            # split into layers by size
+            enc_bboxes_layers, enc_classes_layers = self._unpack_labels(
+                enc_bboxes, enc_classes
+            )
+
+            # interleave enc_bboxes_layers and enc_classes_layers
+            enc_layers = [
+                item
+                for pair in zip(enc_classes_layers, enc_bboxes_layers)
+                for item in pair
+            ]
+
+            bboxes = ops_util.bbox_to_effdet_format(
+                self._device, bboxes, self._image_size
+            )
+            bboxes = dali.fn.pad(
+                bboxes,
+                fill_value=-1,
+                shape=(self._max_instances_per_image, 4),
+                device=self._device,
+            )
+            classes = dali.fn.pad(
+                classes,
+                fill_value=-1,
+                shape=(self._max_instances_per_image,),
+                device=self._device,
+            )
+
+            self._pipe.set_outputs(images, num_positives, bboxes, classes, *enc_layers)
+
+    def _unpack_labels(self, enc_bboxes, enc_classes):
+        # from keras/anchors.py
+
+        enc_bboxes_layers = []
+        enc_classes_layers = []
+
+        count = 0
+        for level in range(self._anchors.min_level, self._anchors.max_level + 1):
+            feat_size = self._anchors.feat_sizes[level]
+            steps = (
+                feat_size["height"]
+                * feat_size["width"]
+                * self._anchors.get_anchors_per_location()
+            )
+
+            enc_bboxes_layers.append(
+                dali.fn.reshape(
+                    dali.fn.slice(
+                        enc_bboxes,
+                        (count, 0),
+                        (steps, 4),
+                        axes=[0, 1],
+                        device=self._device,
+                    ),
+                    shape=[feat_size["height"], feat_size["width"], -1],
+                    device=self._device,
+                )
+            )
```

Comment:
I'm not sure I properly understand this line - what are the anchor and shape arguments?
```suggestion
            enc_bboxes_layers.append(enc_boxes[count:count+steps,:, dali.newaxis])
```
?

Reply:
Didn't work